### PR TITLE
Feature/66 fixes for the upgraded packages

### PR DESCRIPTION
**Pull request by Sachinbisht27**
#66 

### Types of changes

- [ ] Bug fix (a change which fixes an issue)
- [x] New feature (a change which adds functionality)

### Short description of what this resolves:
This implementation is required after upgrading the Flask package to latest.

### Checklist:

- [x] I have performed a self-review of my own code.
- [x] The code follows the style guidelines of this project.
- [x] The code changes are passing the CI checks
- [ ] I have documented my code wherever required.
- [ ] The changes requires a change to the documentation.
- [ ] I have updated the documentation based on the my changes.

### CODE REVIEW

File: api/models/call_log_event.py [hunk 1]
```diff
@@ -1,8 +1,18 @@
...
 from api import db
+from flask_sqlalchemy.query import Query as BaseQuery
+
+
+class CallLogEventQuery(BaseQuery):
+    def call_sid_exist(self, form_data):
+        return (
+            self.filter(CallLogEvent.call_sid == form_data["CallSid"]).first()
+            is not None
+        )
```
Comment (Sachinbisht27) [inline api/models/call_log_event.py:11]: Query to check whether the `call_sid` exists or not in the `call_log_event` table, the query is moved from the `event_service` to the `call_log_event` model.

File: api/models/system_phone.py [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+
+
+class SystemPhoneQuery(BaseQuery):
+    def system_phone_exists(self, url_decoded_system_phone):
+        return (
+            self.filter(
+                SystemPhone.phone.contains(url_decoded_system_phone[-10:])
+            ).first()
+            is not None
+        )
```
Comment (Sachinbisht27) [inline api/models/system_phone.py:13]: Query to check whether the `system_phone_number` exists or not in the `SystemPhone` table, the query is moved from the `event_service` to the `SystemPhone` model.

File: api/services/call_log_event_service.py [hunk 3]
```diff
@@ -1,6 +1,5 @@
 # This file is treated as service layer
-from flask import request
```
Comment (Sachinbisht27) [inline api/services/call_log_event_service.py:2]: Removed as not needed in the file.

File: api/services/handle_event_service.py [hunk 4]
```diff
@@ -1,6 +1,5 @@
 import requests
-from flask import request
```
Comment (Sachinbisht27) [inline api/services/handle_event_service.py:2]: Removed `request` as not needed in the file.

File: api/services/handle_event_service.py [hunk 5]
```diff
@@ -10,18 +9,16 @@
 
 class HandleEventService:
     def handle_event_service(self, form_data):
-        call_sid_exist = db.session.query(
-            db.exists().where(models.CallLogEvent.call_sid == form_data["CallSid"])
-        ).scalar()
+        call_sid_exist = models.CallLogEvent.query.call_sid_exist(form_data)
```
Comment (Sachinbisht27) [inline api/services/handle_event_service.py:12]: Query moved to CallLogEvent model

File: api/services/handle_event_service.py [hunk 6]
```diff
@@ -10,18 +9,16 @@
...
             return
 
         url_decoded_system_phone = requests.utils.unquote(form_data["To"])
 
-        system_phone_exists = db.session.query(
-            db.exists().where(models.SystemPhone.phone == url_decoded_system_phone)
-        ).scalar()
+        system_phone_exists = models.SystemPhone.query.system_phone_exists(
+            url_decoded_system_phone
+        )
```
Comment (Sachinbisht27) [inline api/services/handle_event_service.py:21]: Query moved to SystemPhone model

**Review by vivek-kumar-pandey — APPROVED** [review]
looks fine to me

File: api/services/registration_service.py [hunk 7]
```diff
@@ -8,7 +8,7 @@ class RegistrationService:
     # Create a registartion
     def create_registration(self, data):
         system_phone_data = models.SystemPhone.query.filter_by(
-            phone=data["to_number"]
+            phone=data["to_number"][-10:]
```
Comment (vivek-kumar-pandey) [inline api/services/registration_service.py:11]: Is it fine to check the last 10 digits, as previously we were not doing this?

**Review by vivek-kumar-pandey — COMMENTED** [review]
Just one doubt.

File: api/services/registration_service.py (hunk 7 above)
Comment (Sachinbisht27) [reply api/services/registration_service.py:11]: No, @vivek-kumar-pandey I forgot to remove this I was just testing all the changes. So, added it. Removing now.

**Pull request merged**